### ISSUE TITLE
Nearby pin is not removed, wikidata item's P18 value is not linked.

### ISSUE THREAD

**Issue by neslihanturan** [commons-app/apps-android-commons#3769]
**Summary:** 
We got a recent report about after a nearby upload, wikidata item's P18 value is not edited. Thus nearby pin is not disappeared. Logs will be added if the user send them. 

Uploads:
https://commons.wikimedia.org/wiki/File:%C3%89glise_primitive_de_Carros-Village.jpg
https://commons.wikimedia.org/wiki/File:%C3%89glise_primitive_de_Carros-Village_01.jpg

Wikidata item should have been edited:
https://www.wikidata.org/wiki/Q38396345

**Comment by macgills** [comment]
potential culprit is 
```
    public void handleBadImage(Integer errorCode, UploadItem uploadItem) {
        Timber.d("Handle bad picture with error code %d", errorCode);
        // If location of image and nearby does not match, then set shared preferences to disable wikidata edits
        if (errorCode >= 8) {
            defaultKVStore.putBoolean("Picture_Has_Correct_Location", false);
        }
```

This preference never gets set true, effectively disabling uploading images on the device permanently if the user once tried to upload a picture with a distant location to a Place, unless the user wipes their data.

>if the user once tried to upload a picture with a distant location to a Place

This could also be not true, the error handling is not predictable and an issue to improve it has been open a while #3500

**Comment by nicolas-raoul** [comment]
I uploaded a lot of pictures via Nearby yesterday, and none of them was correctly linked from P18.
Example log:
```
2020-09-06 21:32:14,195     [DEBUG] [file-logging-thread-1] [NearbyPlaces] : 61 results at radius: 1.000000
2020-09-06 21:32:14,196     [DEBUG] [file-logging-thread-1] [NearbyController] : Sorting places by distance...
2020-09-06 21:32:15,122     [DEBUG] [file-logging-thread-1] [NearbyParentFragment] : Updates map camera to track user position
2020-09-06 21:32:15,128     [DEBUG] [file-logging-thread-1] [NearbyParentFragment] : Adds current location marker
2020-09-06 21:32:15,129     [DEBUG] [file-logging-thread-1] [NearbyParentFragment] : Map is centered to place
2020-09-06 21:32:15,160     [DEBUG] [file-logging-thread-1] [NearbyParentFragment] : Adds current location marker
2020-09-06 21:32:15,199     [DEBUG] [file-logging-thread-1] [NearbyParentFragment] : Adds current location marker
2020-09-06 21:32:29,548     [DEBUG] [file-logging-thread-1] [NearbyParentFragment] : Location service manager unregistered and removed
2020-09-06 21:32:30,138     [DEBUG] [file-logging-thread-1] [NearbyParentFragment] : Checking permission and perfoming action
2020-09-06 21:32:30,149     [DEBUG] [file-logging-thread-1] [NearbyParentFragmentPresenter] : Presenter updates map and list
2020-09-06 21:32:30,154     [DEBUG] [file-logging-thread-1] [NearbyParentFragmentPresenter] : LOCATION_SIGNIFICANTLY_CHANGED
2020-09-06 21:32:30,155     [DEBUG] [file-logging-thread-1] [NearbyParentFragment] : Location service manager added and registered
2020-09-06 21:32:30,156     [DEBUG] [file-logging-thread-1] [NearbyController] : Loading attractions near lat/lng: (35.6452214,139.7172868)
2020-09-06 21:32:30,227     [DEBUG] [file-logging-thread-1] [NearbyPlaces] : 61 results at radius: 1.000000
2020-09-06 21:32:30,229     [DEBUG] [file-logging-thread-1] [NearbyController] : Sorting places by distance...
2020-09-06 21:32:30,568     [DEBUG] [file-logging-thread-1] [NearbyParentFragment] : Updates map camera to track user position
2020-09-06 21:32:30,569     [DEBUG] [file-logging-thread-1] [NearbyParentFragment] : Adds current location marker
2020-09-06 21:32:30,586     [DEBUG] [file-logging-thread-1] [NearbyParentFragment] : Adds current location marker
2020-09-06 21:32:30,656     [DEBUG] [file-logging-thread-1] [MainActivity] : Number of notifications is 0
2020-09-06 21:32:42,398     [DEBUG] [file-logging-thread-1] [LocationServiceManager] : on location changed
2020-09-06 21:32:42,399     [DEBUG] [file-logging-thread-1] [NearbyParentFragment] : Location slightly changed
2020-09-06 21:32:42,400     [DEBUG] [file-logging-thread-1] [NearbyParentFragmentPresenter] : Presenter updates map and list
2020-09-06 21:32:42,400     [DEBUG] [file-logging-thread-1] [NearbyController] : Loading attractions near lat/lng: (35.645465014110314,139.71753317945817)
2020-09-06 21:32:42,406     [DEBUG] [file-logging-thread-1] [NearbyParentFragmentPresenter] : Means location changed slightly
2020-09-06 21:32:42,407     [DEBUG] [file-logging-thread-1] [NearbyParentFragment] : Current location marker invisible
2020-09-06 21:32:42,407     [DEBUG] [file-logging-thread-1] [LocationServiceManager] : gps's status changed to 1
2020-09-06 21:32:45,561     [DEBUG] [file-logging-thread-1] [NearbyPlaces] : 64 results at radius: 1.000000
2020-09-06 21:32:45,564     [DEBUG] [file-logging-thread-1] [NearbyController] : Sorting places by distance...
2020-09-06 21:32:45,568     [DEBUG] [file-logging-thread-1] [NearbyNotificationCardView] : Update nearby card notification content
2020-09-06 21:33:01,555     [DEBUG] [file-logging-thread-1] [NearbyParentFragment] : Map is centered to place
2020-09-06 21:33:05,127     [DEBUG] [file-logging-thread-1] [CommonPlaceClickActions$onGalleryClicked] : Gallery button tapped. Image title: Embassy of the Czech Republic, TokyoImage desc: embassy
2020-09-06 21:33:05,129     [DEBUG] [file-logging-thread-1] [CommonPlaceClickActions] : Store place object Place{name='Embassy of the Czech Republic, Tokyo', label='UNKNOWN', longDescription='embassy', location='lat/lng: (35.651938,139.716795)', category='', distance='748m', siteLinks='Sitelinks{wikipediaLink='', commonsLink='', wikidataLink='http://www.wikidata.org/entity/Q61877435'}', pic='', destroyed=''}
2020-09-06 21:33:05,162     [DEBUG] [file-logging-thread-1] [NearbyParentFragment] : Location service manager unregistered and removed
2020-09-06 21:33:14,509     [DEBUG] [file-logging-thread-1] [MainActivity] : Intent { dat=content://com.android.providers.media.documents/document/image:66628 flg=0x1 }
2020-09-06 21:33:14,618     [DEBUG] [file-logging-thread-1] [NearbyParentFragment] : Checking permission and perfoming action
2020-09-06 21:33:14,626     [DEBUG] [file-logging-thread-1] [NearbyParentFragmentPresenter] : Presenter updates map and list
2020-09-06 21:33:14,627     [DEBUG] [file-logging-thread-1] [NearbyParentFragmentPresenter] : LOCATION_SIGNIFICANTLY_CHANGED
2020-09-06 21:33:14,627     [DEBUG] [file-logging-thread-1] [NearbyController] : Loading attractions near lat/lng: (35.645465014110314,139.71753317945817)
2020-09-06 21:33:14,628     [DEBUG] [file-logging-thread-1] [NearbyParentFragment] : Location service manager added and registered
2020-09-06 21:33:14,636     [DEBUG] [file-logging-thread-1] [NearbyParentFragment] : Location service manager unregistered and removed
2020-09-06 21:33:14,710     [DEBUG] [file-logging-thread-1] [UploadActivity] : Received intent Intent { act=internalImageUploads cmp=fr.free.nrw.commons/.upload.UploadActivity (has extras) } with action internalImageUploads
2020-09-06 21:33:14,711     [INFO ] [file-logging-thread-1] [UploadActivity] : Received multiple upload 1
2020-09-06 21:33:14,791     [DEBUG] [file-logging-thread-1] [UploadModel] : File created date is 1599414578000
2020-09-06 21:33:14,794     [DEBUG] [file-logging-thread-1] [FileMetadataUtils] : Retuning tags for pref:Author
2020-09-06 21:33:14,795     [DEBUG] [file-logging-thread-1] [FileProcessor] : Checking for tag: Artist
2020-09-06 21:33:14,795     [DEBUG] [file-logging-thread-1] [FileProcessor] : Checking for tag: CameraOwnerName
2020-09-06 21:33:14,796     [DEBUG] [file-logging-thread-1] [FileMetadataUtils] : Retuning tags for pref:Copyright
2020-09-06 21:33:14,796     [DEBUG] [file-logging-thread-1] [FileProcessor] : Checking for tag: Copyright
2020-09-06 21:33:14,797     [DEBUG] [file-logging-thread-1] [FileMetadataUtils] : Retuning tags for pref:Camera Model
2020-09-06 21:33:14,798     [DEBUG] [file-logging-thread-1] [FileProcessor] : Checking for tag: Make
2020-09-06 21:33:14,798     [DEBUG] [file-logging-thread-1] [FileProcessor] : Exif tag Make with value kotlin.Unit redacted.
2020-09-06 21:33:14,798     [DEBUG] [file-logging-thread-1] [FileProcessor] : Checking for tag: Model
2020-09-06 21:33:14,798     [DEBUG] [file-logging-thread-1] [FileProcessor] : Exif tag Model with value kotlin.Unit redacted.
2020-09-06 21:33:14,799     [DEBUG] [file-logging-thread-1] [FileMetadataUtils] : Retuning tags for pref:Lens Model
2020-09-06 21:33:14,800     [DEBUG] [file-logging-thread-1] [FileProcessor] : Checking for tag: LensMake
2020-09-06 21:33:14,801     [DEBUG] [file-logging-thread-1] [FileProcessor] : Checking for tag: LensModel
2020-09-06 21:33:14,802     [DEBUG] [file-logging-thread-1] [FileProcessor] : Checking for tag: LensSpecification
2020-09-06 21:33:14,803     [DEBUG] [file-logging-thread-1] [FileMetadataUtils] : Retuning tags for pref:Serial Numbers
2020-09-06 21:33:14,804     [DEBUG] [file-logging-thread-1] [FileProcessor] : Checking for tag: BodySerialNumber
2020-09-06 21:33:14,805     [DEBUG] [file-logging-thread-1] [FileProcessor] : Checking for tag: LensSerialNumber
2020-09-06 21:33:14,806     [DEBUG] [file-logging-thread-1] [FileMetadataUtils] : Retuning tags for pref:Software
2020-09-06 21:33:14,807     [DEBUG] [file-logging-thread-1] [FileProcessor] : Checking for tag: Software
2020-09-06 21:33:14,808     [DEBUG] [file-logging-thread-1] [FileProcessor] : Exif tag Software with value kotlin.Unit redacted.
2020-09-06 21:33:14,819     [DEBUG] [file-logging-thread-1] [FileProcessor] : Calling GPSExtractor
2020-09-06 21:33:14,819     [DEBUG] [file-logging-thread-1] [ImageCoordinates] : EXIF data has location info
2020-09-06 21:33:14,821     [DEBUG] [file-logging-thread-1] [CategoryApi] : URL: https://commons.wikimedia.org/w/api.php?action=query&prop=categories%7Ccoordinates%7Cpageprops&format=json&clshow=%21hidden&coprop=type%7Cname%7Cdim%7Ccountry%7Cregion%7Cglobe&codistancefrompoint=35.65177479972222%7C139.7171759&generator=geosearch&ggscoord=35.65177479972222%7C139.7171759&ggsradius=10000&ggslimit=10&ggsnamespace=6&ggsprop=type%7Cname%7Cdim%7Ccountry%7Cregion%7Cglobe&ggsprimary=all&formatversion=2
2020-09-06 21:33:14,882     [DEBUG] [file-logging-thread-1] [MediaLicenseFragment] : Position: 0 CC0
2020-09-06 21:33:15,169     [INFO ] [file-logging-thread-1] [UploadMediaDetailAdapter$ViewHolder] : descItemEditText:androidx.appcompat.widget.AppCompatEditText{b8409db VFED..CL. ......I. 0,0-0,0 #7f0900d5 app:id/description_item_edit_text}
2020-09-06 21:33:15,171     [DEBUG] [file-logging-thread-1] [UploadMediaDetailAdapter$ViewHolder] : UploadMediaDetail is UploadMediaDetail(languageCode=en, descriptionText=embassy, captionText=Embassy of the Czech Republic, Tokyo)
2020-09-06 21:33:15,502     [DEBUG] [file-logging-thread-1] [MainActivity] : Number of notifications is 0
2020-09-06 21:33:15,533     [DEBUG] [file-logging-thread-1] [UploadMediaDetailAdapter$ViewHolder] : Description language code is: en
2020-09-06 21:33:16,086     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:16,089     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:16,091     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:16,093     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:16,096     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:16,098     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:16,101     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:16,102     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:16,104     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:16,106     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:16,108     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:16,110     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:16,112     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:16,113     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:16,116     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:16,117     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:16,121     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:16,121     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:16,123     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:16,124     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:16,126     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:16,128     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:16,129     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:16,131     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:16,132     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:23,119     [DEBUG] [file-logging-thread-1] [ImageProcessingService] : Current image quality is -2
2020-09-06 21:33:23,121     [DEBUG] [file-logging-thread-1] [ImageProcessingService] : Checking the validity of image
2020-09-06 21:33:23,122     [DEBUG] [file-logging-thread-1] [ImageProcessingService] : Checking for duplicate image /data/user/0/fr.free.nrw.commons/cache/CommonsContributions/37512aaf-6ba6-4f66-9448-21e72b392c53.jpg
2020-09-06 21:33:23,123     [DEBUG] [file-logging-thread-1] [ImageProcessingService] : Checking for image geolocation /data/user/0/fr.free.nrw.commons/cache/CommonsContributions/37512aaf-6ba6-4f66-9448-21e72b392c53.jpg
2020-09-06 21:33:23,125     [DEBUG] [file-logging-thread-1] [ImageProcessingService] : Checking for dark image /data/user/0/fr.free.nrw.commons/cache/CommonsContributions/37512aaf-6ba6-4f66-9448-21e72b392c53.jpg
2020-09-06 21:33:23,126     [DEBUG] [file-logging-thread-1] [ImageProcessingService] : Checking for image title [UploadMediaDetail(languageCode=en, descriptionText=embassy, captionText=Embassy of the Czech Republic, Tokyo)]
2020-09-06 21:33:23,133     [DEBUG] [file-logging-thread-1] [ImageCoordinates] : EXIF data has location info
2020-09-06 21:33:23,137     [INFO ] [file-logging-thread-1] [FileUtils] : File SHA1: 1558636d25823c9ed78e9ca0c911b91b7d8f4553
2020-09-06 21:33:23,139     [DEBUG] [file-logging-thread-1] [ImageUtils] : Comparing geolocation of file with nearby place location
2020-09-06 21:33:23,526     [DEBUG] [file-logging-thread-1] [ImageProcessingService] : Result for valid title is false
2020-09-06 21:33:23,543     [DEBUG] [file-logging-thread-1] [ImageUtils] : Checking image darkness took 413 ms.
2020-09-06 21:33:23,651     [DEBUG] [file-logging-thread-1] [ImageProcessingService] : Result for duplicate image false
2020-09-06 21:33:23,654     [DEBUG] [file-logging-thread-1] [ImageProcessingService] : duplicate: 0, geo: 0, dark: 0, title: 0fbmd:0exif:0
2020-09-06 21:33:32,386     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:32,388     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:32,390     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:32,392     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:32,394     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:32,405     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:32,406     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:32,407     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:32,407     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:32,408     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:32,409     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:32,410     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:32,411     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:32,411     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:32,412     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:32,413     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:32,414     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:32,415     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:32,416     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:32,418     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:32,418     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:32,419     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:32,419     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:32,420     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:32,420     [DEBUG] [file-logging-thread-1] [CategoriesModel] : Previous year: 2019
2020-09-06 21:33:46,164     [DEBUG] [file-logging-thread-1] [UploadController] : UploadController/doInBackground, contribution.getLocalUri():/data/user/0/fr.free.nrw.commons/cache/CommonsContributions/37512aaf-6ba6-4f66-9448-21e72b392c53.jpg
2020-09-06 21:33:46,203     [DEBUG] [file-logging-thread-1] [UploadService] : Before execution!
2020-09-06 21:33:46,226     [DEBUG] [file-logging-thread-1] [UploadClient] : Chunk info
2020-09-06 21:33:46,603     [DEBUG] [file-logging-thread-1] [NearbyParentFragment] : Checking permission and perfoming action
2020-09-06 21:33:46,608     [DEBUG] [file-logging-thread-1] [NearbyParentFragmentPresenter] : Presenter updates map and list
2020-09-06 21:33:46,616     [DEBUG] [file-logging-thread-1] [NearbyParentFragmentPresenter] : LOCATION_SIGNIFICANTLY_CHANGED
2020-09-06 21:33:46,617     [DEBUG] [file-logging-thread-1] [NearbyParentFragment] : Location service manager added and registered
2020-09-06 21:33:46,619     [DEBUG] [file-logging-thread-1] [NearbyController] : Loading attractions near lat/lng: (35.645465014110314,139.71753317945817)
2020-09-06 21:33:46,704     [DEBUG] [file-logging-thread-1] [NearbyPlaces] : 64 results at radius: 1.000000
2020-09-06 21:33:46,714     [DEBUG] [file-logging-thread-1] [NearbyController] : Sorting places by distance...
2020-09-06 21:33:47,332     [DEBUG] [file-logging-thread-1] [NearbyParentFragment] : Adds current location marker
2020-09-06 21:33:47,358     [DEBUG] [file-logging-thread-1] [NearbyParentFragment] : Adds current location marker
2020-09-06 21:33:47,401     [DEBUG] [file-logging-thread-1] [MainActivity] : Number of notifications is 0
2020-09-06 21:33:49,044     [DEBUG] [file-logging-thread-1] [MainActivity] : Contributions tab selected
2020-09-06 21:33:49,054     [DEBUG] [file-logging-thread-1] [MainActivity] : Contributions fragment notifications menu item is visible
2020-09-06 21:33:49,203     [DEBUG] [file-logging-thread-1] [LocationServiceManager] : on location changed
2020-09-06 21:33:49,204     [DEBUG] [file-logging-thread-1] [NearbyParentFragment] : Location slightly changed
2020-09-06 21:33:49,205     [DEBUG] [file-logging-thread-1] [NearbyParentFragmentPresenter] : Presenter updates map and list
2020-09-06 21:33:49,208     [DEBUG] [file-logging-thread-1] [NearbyParentFragmentPresenter] : Means location changed slightly
2020-09-06 21:33:49,209     [DEBUG] [file-logging-thread-1] [NearbyParentFragment] : Current location marker visible
2020-09-06 21:33:49,212     [DEBUG] [file-logging-thread-1] [NearbyController] : Loading attractions near lat/lng: (35.6452029,139.717282)
2020-09-06 21:33:49,213     [DEBUG] [file-logging-thread-1] [NearbyParentFragment] : Adds current location marker
2020-09-06 21:33:49,450     [DEBUG] [file-logging-thread-1] [NearbyPlaces] : 61 results at radius: 1.000000
2020-09-06 21:33:49,461     [DEBUG] [file-logging-thread-1] [NearbyController] : Sorting places by distance...
2020-09-06 21:33:49,486     [DEBUG] [file-logging-thread-1] [NearbyNotificationCardView] : Update nearby card notification content
2020-09-06 21:34:04,810     [DEBUG] [file-logging-thread-1] [LocationServiceManager] : on location changed
2020-09-06 21:34:04,830     [DEBUG] [file-logging-thread-1] [NearbyParentFragment] : Location slightly changed
2020-09-06 21:34:04,830     [DEBUG] [file-logging-thread-1] [NearbyParentFragmentPresenter] : Presenter updates map and list
2020-09-06 21:34:04,833     [DEBUG] [file-logging-thread-1] [NearbyParentFragmentPresenter] : Means location changed slightly
2020-09-06 21:34:04,834     [DEBUG] [file-logging-thread-1] [NearbyParentFragment] : Current location marker visible
2020-09-06 21:34:04,844     [DEBUG] [file-logging-thread-1] [NearbyParentFragment] : Adds current location marker
2020-09-06 21:34:04,844     [DEBUG] [file-logging-thread-1] [LocationServiceManager] : gps's status changed to 1
2020-09-06 21:34:26,679     [DEBUG] [file-logging-thread-1] [UploadService] : Stash upload response 1 is StashUploadResult(state=SUCCESS, fileKey=17qtuek5idt8.3053lb.27103.jpg)
2020-09-06 21:34:26,681     [DEBUG] [file-logging-thread-1] [UploadService] : making sure of uniqueness of name: Embassy of the Czech Republic, Tokyo.jpg
2020-09-06 21:34:30,622     [DEBUG] [file-logging-thread-1] [UploadService] : Stash upload response 2 is UploadResult(result=Success, filekey=null, offset=0, filename=Embassy_of_the_Czech_Republic,_Tokyo.jpg)
2020-09-06 21:34:30,625     [DEBUG] [file-logging-thread-1] [WikidataEditService] : Image location and nearby place location mismatched, so Wikidata item won't be edited
2020-09-06 21:34:31,061     [DEBUG] [file-logging-thread-1] [WikidataEditService] : EntityId for image was received successfully: 93887019
2020-09-06 21:34:32,628     [DEBUG] [file-logging-thread-1] [WikidataEditService] : DEPICTS property was set successfully for 93887019
2020-09-06 21:34:32,639     [DEBUG] [file-logging-thread-1] [WikidataEditService] : edit response: true
2020-09-06 21:34:34,358     [DEBUG] [file-logging-thread-1] [WikidataEditService] : DEPICTS property was set successfully for 93887019
2020-09-06 21:34:34,361     [DEBUG] [file-logging-thread-1] [WikidataEditService] : edit response: true
2020-09-06 21:34:36,083     [DEBUG] [file-logging-thread-1] [WikidataEditService] : Caption successfully set, revision id = org.wikipedia.dataclient.mwapi.MwPostResponse@ff1a33f[success=1,errors=<null>]
2020-09-06 21:34:36,084     [DEBUG] [file-logging-thread-1] [WikidataEditService] : edit response: true
2020-09-06 21:34:39,032     [DEBUG] [file-logging-thread-1] [LocationServiceManager] : on location changed
2020-09-06 21:34:39,049     [DEBUG] [file-logging-thread-1] [NearbyParentFragment] : Location slightly changed
2020-09-06 21:34:39,050     [DEBUG] [file-logging-thread-1] [NearbyParentFragmentPresenter] : Presenter updates map and list
2020-09-06 21:34:39,050     [DEBUG] [file-logging-thread-1] [NearbyParentFragmentPresenter] : Means location changed slightly
2020-09-06 21:34:39,050     [DEBUG] [file-logging-thread-1] [NearbyParentFragment] : Current location marker visible
2020-09-06 21:34:39,050     [DEBUG] [file-logging-thread-1] [LocationServiceManager] : gps's status changed to 1
2020-09-06 21:34:39,051     [DEBUG] [file-logging-thread-1] [NearbyParentFragment] : Adds current location marker
2020-09-06 21:34:43,026     [DEBUG] [file-logging-thread-1] [LocationServiceManager] : gps's status changed to 1
2020-09-06 21:35:08,214     [DEBUG] [file-logging-thread-1] [LocationServiceManager] : on location changed
2020-09-06 21:35:08,217     [DEBUG] [file-logging-thread-1] [NearbyParentFragment] : Location slightly changed
2020-09-06 21:35:08,218     [DEBUG] [file-logging-thread-1] [NearbyParentFragmentPresenter] : Presenter updates map and list
2020-09-06 21:35:08,219     [DEBUG] [file-logging-thread-1] [NearbyParentFragmentPresenter] : Means location changed slightly
2020-09-06 21:35:08,220     [DEBUG] [file-logging-thread-1] [NearbyParentFragment] : Current location marker visible
2020-09-06 21:35:08,221     [DEBUG] [file-logging-thread-1] [NearbyParentFragment] : Adds current location marker
2020-09-06 21:35:08,227     [DEBUG] [file-logging-thread-1] [LocationServiceManager] : gps's status changed to 1
2020-09-06 21:35:18,463     [DEBUG] [file-logging-thread-1] [NearbyParentFragment] : Location service manager unregistered and removed
```
The problematic part is `Image location and nearby place location mismatched, so Wikidata item won't be edited`. It is clearly wrong as the image https://commons.wikimedia.org/wiki/File:Embassy_of_the_Czech_Republic,_Tokyo.jpg has coordinates only 43 meters from the Wikidata item:
![Screen Shot 2020-09-07 at 11 41 02](https://user-images.githubusercontent.com/99590/92343732-7e07f980-f0ff-11ea-9db2-3ee69014a0ab.png)
macgills' comment above is on point. I think the result of the check should be passed along (maybe in the UploadItem?), rather than stored as a preference.

**Comment by ashishkumar468** [comment]
Possible Duplicate of #3856 , which has been fixed and merged in 2.13-release

**Comment by nicolas-raoul** [comment]
Indeed, it works in branch 2.13-release, thanks a lot! :-)
